File: day7/same_tree.java
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        // TreeNode temp=p;
        // TreeNode tempr=q;
        // if(p==null || q==null) return false;
        // while(p.left!=null && q.left!=null){
        //     if(p.val!=q.val) return false;
        //     p=p.left;
        //     q=q.left;
        // }
        //  while(p.right!=null && q.right!=null){
        //     if(p.val!=q.val) return false;
        //     p=p.right;
        //     q=q.right;
        // }
        // return true;

        if(p==null && q==null) return true;
        if(p==null || q==null || p.val!=q.val) return false;
        return isSameTree(p.left,q.left) && isSameTree(p.right,q.right);
    }
}

File: same tree.py
#100. Same Tree
class Solution:
    def isSameTree(self, p, q):
        # Both nodes are None
        if not p and not q:
            return True

        # One is None and the other is not
        if not p or not q:
            return False

        # Values are different
        if p.val != q.val:
            return False

        # Check left and right subtrees
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)